nav2_docking/opennav_docking/include/opennav_docking/simple_charging_dock.hpp

```diff
@@ -149,6 +161,28 @@ class SimpleChargingDock : public opennav_docking_core::ChargingDock
 
   rclcpp_lifecycle::LifecycleNode::SharedPtr node_;
   std::shared_ptr<tf2_ros::Buffer> tf2_buffer_;
+
+  // Detector control parameters
+  std::string detector_service_name_;
+  double detector_service_timeout_{5.0};
+  bool subscribe_toggle_{false};
+
+  // Client used to call the Trigger service
+  nav2_util::ServiceClient<
```

Comment:
Rebase  / merge in changes. This class is now different (nav2::ServiceClient) and should be gotten from the `nav2::LifecycleNode`'s `create_client`. See https://docs.nav2.org/migration/Kilted.html#new-nav2-ros-common-nav2-lifecycle-node